### PR TITLE
Adds reflection and conversions from common types

### LINKED ISSUE #53: Investigate the possibility of automatically converting a struct into a matioCpp struct using reflection
Related links:

https://stackoverflow.com/questions/63467178/modern-way-to-access-struct-fields-by-name-in-c

In particular, I found pretty interesting https://github.com/garbageslam/visit_struct

The main drawback is that the struct needs to be modified, specifying which entry can be accessed. On the other hand, this might be useful since some types cannot be automatically converted into ``matioCpp`` variables. In addition, the options that do not need to modify the struct, usually cannot get the names of the fields. See https://bytemaster.github.io/boost_reflect/ and https://github.com/apolukhin/magic_get

### PR DESCRIPTION
**Pull request by S-Dafarra**
I also fixed the CI failures and some issues with the license

Fix #53

### CODE REVIEW

**Comment by S-Dafarra** [comment]
@traversaro @GiulioRomualdi friendly ping

File: include/matioCpp/Element.h [hunk 1]
```diff
@@ -105,6 +105,16 @@ class matioCpp::Element : public matioCpp::Variable
...
 
+    /**
+     * @brief Get this Vector as a Span
+     */
+    matioCpp::Span<element_type> toSpan();
+
+    /**
+     * @brief Get this Vector as a Span (const version)
+     */
+    const matioCpp::Span<const element_type> toSpan() const;
```
Comment (traversaro) [inline include/matioCpp/Element.h:116]: Nitpick, i.e. feel free to ignore: for methods that just provide a different "view" of the existing object/memory, I tipically find "as<>" methods (so in this case `asSpan`) more descriptive.

**Review by traversaro — APPROVED** [review]
Minor comments.

File: include/matioCpp/Element.h (hunk 1 above)
Comment (S-Dafarra) [reply include/matioCpp/Element.h:116]: I have used ``as*`` for the downcasting of a ``Variable`` object to its derived, while I used ``to*`` for other types (like [``toMatio`` ](https://github.com/ami-iit/matio-cpp/blob/1555d2451dbc56665b89d82028cd3225b24d0e82/include/matioCpp/Variable.h#L283) and ``toSpan``). To be honest, I am not sure this difference was completely intentional at the beginning, but I would keep it as is for the time being. The main reason I added ``toSpan`` to ``Element`` was to have the possibility of using that method in templated functions accepting also ``Vector`` and ``MultiDimensionalArray``, where the ``toSpan`` method is already defined.

**Comment by S-Dafarra** [comment]
Thanks a lot @traversaro. Merging

**Pull request merged**